compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2187,6 +2199,17 @@ private ParserRuleContext getNextRuleForCloseBrace(int nextLookahead) {
                     return ParserRuleContext.STATEMENT;
                 } else if (parentCtx == ParserRuleContext.NAMED_WORKER_DECL) {
                     endContext(); // end named-worker
+                    parentCtx = getParentContext();
+                    if (parentCtx == ParserRuleContext.FORK_STMT) {
+                        nextToken = this.tokenReader.peek(nextLookahead);
+                        switch (nextToken.kind) {
+                            case AT_TOKEN:
+                            case WORKER_KEYWORD:
+                                return ParserRuleContext.NAMED_WORKER_DECL;
+                            default:
+                                return ParserRuleContext.CLOSE_BRACE;
+                        }
+                    }
```

Comment:
Else throw an error. Then we know the parser has come to an invalid state, and easy to debug :)

Reply:
else for the if at L#2203?, its ok if the flow doesnt go through that because when doing a normal worker declaration the flow is to return a statement at that point.